roles/etcd/defaults/main.yml

```diff
@@ -1,14 +1,53 @@
 ---
 # role variable for the etcd service
 
+etcd_version: v2.3.8
+etcd_cluster_name: contiv
+
+etcd_proxy: "{{ etcd_proxies_group in groups
+                 and inventory_hostname in groups[etcd_proxies_group] }}"
+etcd_member: "{{ etcd_members_group in groups
+                 and inventory_hostname in groups[etcd_members_group] }}"
+etcd_service_name: "etcd{{ etcd_proxy|bool|ternary('-proxy','') }}-{{
+    etcd_cluster_name}}"
+etcd_docker_name: "{{ etcd_service_name }}"
+
+etcd_node_name: "{{ inventory_hostname_short }}"
 etcd_client_port1: 2379
 etcd_client_port2: 4001
+
 etcd_peer_port1: 2380
-etcd_peer_port2: 7001
-etcd_peers_group: "service-master"
-etcd_rule_comment: "contiv_etcd traffic"
-etcd_version: "v2.3.8"
+
+etcd_members_group: netplugin-master
+etcd_proxies_group: netplugin-worker
+etcd_initial_cluster_token: "etcd-{{ etcd_cluster_name }}"
+
+etcd_rule_comment: "{{ etcd_cluster_name }}_etcd traffic"
+
 etcd_heartbeat_interval: 1000
 etcd_election_timeout: 10000
-etcd_data_dir: /var/lib/etcd
+
+etcd_data_dir: /var/lib/etcd/data
+etcd_wal_dir: /var/lib/etcd/wal
+
 etcd_reset_state: false
+
+# Etcd has "advertise" urls for the other nodes to use if it's behind a proxy
+# Also, it needs to "listen" on IPs and ports for peers to talk to
+etcd_peer_advertise_interface: "{{ ansible_default_ipv4.interface }}"
+etcd_peer_advertise_address: "{{
```

Comment:
I think we should just retire `etcd_peer_advertise_interface` and use list of ips for  `etcd_peer_advertise_address`, this way if the default interface has more than one ips, it will have problems

Reply:
inv can set `etcd_peer_advertise_address` in that scenario, this makes it easy for VMs (like molecule) where the interface is the same across all the VMs, otherwise that ugly ip fetch is pushed up higher level

Reply:
i think molecule is special cases and usually it should just advertise  `ansible_default_ipv4.address`, this also avoids overridded `etcd_peer_advertise_address` doesn't match `etcd_peer_advertise_interface`,  also it's a little strange to say a service listen on a interface, but listen on a ip:port

Reply:
default ipv4 in vbox is 10.0.2.15 which is not useful, have to target the second interface

Reply:
also contiv provides a control_interface, does not provide ip addresses

Reply:
`control_interface` is one of things i think we need to remove from etcd dependency i think, it was used in many places but most places are used to get it's ip address like here. 
I think vbox is a special case we need to workaround by set `etcd_peer_advertise_address` and leave default as `ansible_default_ipv4.address`.
This is not a blocker for refactoring etcd role, but this is going to be helpful moving forward

Reply:
good ideas, just there's more than enough scope with this change